### PR TITLE
feat: vote config

### PR DESCRIPTION
**Pull request by catngh**
**What does this PR do?**

- Add APIs to config vote channel

`GET` and `POST` response
image

### CODE REVIEW

File: pkg/entities/configs.go [hunk 1]
```diff
@@ -19,6 +19,7 @@ func (e *Entity) GetGmConfig(guildID string) (*model.GuildConfigGmGn, error) {
 		if err == gorm.ErrRecordNotFound {
 			return nil, nil
 		}
+		e.log.Error(err, "[Entity][GetGmConfig] repo.GuildConfigGmGn.GetByGuildID failed")
```
Comment (trkhoi) [inline pkg/entities/configs.go:22]: e.log.Field(logger.Fields{"guildID": guildID}).Error(...)

File: pkg/entities/configs.go [hunk 2]
```diff
@@ -30,6 +31,7 @@ func (e *Entity) UpsertGmConfig(req request.UpsertGmConfigRequest) error {
 		GuildID:   req.GuildID,
 		ChannelID: req.ChannelID,
 	}); err != nil {
+		e.log.Error(err, "[Entity][UpsertGmConfig] repo.GuildConfigGmGn.UpsertOne failed")
```
Comment (trkhoi) [inline pkg/entities/configs.go:34]: e.log.Field(logger.Fields{"guildID": guildID, "channelID": channelID}).Error(...)

File: pkg/entities/configs.go [hunk 3]
```diff
@@ -42,6 +44,7 @@ func (e *Entity) GetWelcomeChannelConfig(guildID string) (*model.GuildConfigWelc
 		if err == gorm.ErrRecordNotFound {
 			return nil, nil
 		}
+		e.log.Error(err, "[Entity][GetWelcomeChannelConfig] repo.GuildConfigWelcomeChannel.GetByGuildID failed")
```
Comment (trkhoi) [inline pkg/entities/configs.go:47]: same as above

File: pkg/entities/configs.go [hunk 4]
```diff
@@ -58,6 +61,7 @@ func (e *Entity) UpsertWelcomeChannelConfig(req request.UpsertWelcomeConfigReque
 		WelcomeMessage: req.WelcomeMsg,
 	})
 	if err != nil {
+		e.log.Error(err, "[Entity][UpsertWelcomeChannelConfig] repo.GuildConfigWelcomeChannel.UpsertOne failed")
```
Comment (trkhoi) [inline pkg/entities/configs.go:64]: same as above

File: pkg/entities/configs.go [hunk 5]
```diff
@@ -68,6 +72,42 @@ func (e *Entity) DeleteWelcomeChannelConfig(req request.DeleteWelcomeConfigReque
 	if err := e.repo.GuildConfigWelcomeChannel.DeleteOne(&model.GuildConfigWelcomeChannel{
 		GuildID: req.GuildID,
 	}); err != nil {
+		e.log.Error(err, "[Entity][DeleteWelcomeChannelConfig] repo.GuildConfigWelcomeChannel.DeleteOne failed")
```
Comment (trkhoi) [inline pkg/entities/configs.go:75]: same as above

File: pkg/entities/configs.go [hunk 6]
```diff
@@ -68,6 +72,42 @@ func (e *Entity) DeleteWelcomeChannelConfig(req request.DeleteWelcomeConfigReque
...
+	return nil
+}
+
+func (e *Entity) GetVoteChannelConfig(guildID string) (*model.GuildConfigVoteChannel, error) {
+	config, err := e.repo.GuildConfigVoteChannel.GetByGuildID(guildID)
+	if err != nil {
+		if err == gorm.ErrRecordNotFound {
+			return nil, nil
+		}
+		e.log.Error(err, "[Entity][GetVoteChannelConfig] repo.GuildConfigVoteChannel.GetByGuildID failed")
```
Comment (trkhoi) [inline pkg/entities/configs.go:88]: log.Field(logger.Fields{"guildID": guildID}).Error(...)

**Review by trkhoi — CHANGES_REQUESTED** [review]

**Review by tuanddd — APPROVED** [review]

**Review by trkhoi — APPROVED** [review]

**Pull request merged**